scanner/src/main/kotlin/Scanner.kt

```diff
@@ -228,6 +228,9 @@ class Scanner(
                             )
                         )
                     }
+
+                    // If resolving the nested repositories failed, we can still scan the main repository.
+                    controller.addNestedProvenance(provenance, NestedProvenance(provenance, emptyMap()))
```

Comment:
I deliberately did not catch or handle those errors so far because they indicate bugs, this helped fixing issues in the past. Can you provide a stacktrace for the exception to see where exactly it did occur? I suspect that the root cause might be in `getNestedScanResultsByPackage()` which does not filter packages with provenance resolution errors.

Reply:
For the stack trace, please see https://github.com/oss-review-toolkit/ort/issues/5899.

Reply:
Ok, then it is indeed `getNestedScanResultsByPackage()`. I think this function should be changed to not try to build a scan result for a package where provenance resolution failed by calling `hasCompleteScanResult()` on the packages ((after changing `hasCompleteScanResult()` to take an `Identifier` instead of a `Package` as argument)).

Reply:
IIUC you propose the following: In case nested provenance resolution failed AND provenance resolution of main repository succeeded, do not return any scan result even if the main repository has been scanned. Can you explain "why" you think it is better to not use the scan result of the main repository?

Reply:
> Ok, then it is indeed `getNestedScanResultsByPackage()`. I think this function should be changed to not try to build a scan result for a package where provenance resolution failed by calling `hasCompleteScanResult()` on the packages (after changing `hasCompleteScanResult()` to take an `Identifier` instead of a `Package` as argument).

I tried your approach adjusting `hasCompleteScanResult()`, but failed as that has `scanner` as second parameter which seems to me to not fit the use case. So, I've researched a bit further and found a solution which I hope aligns with your idea. 
I made it a separate PR, https://github.com/oss-review-toolkit/ort/pull/5933, as I didn't feel ready to scratch this one yet.
However, we can use that PR as replacement if we consider it to be ok.

Reply:
> IIUC you propose the following: In case nested provenance resolution failed AND provenance resolution of main repository succeeded, do not return any scan result even if the main repository has been scanned. Can you explain "why" you think it is better to not use the scan result of the main repository?

If I'm not wrong the main repository is not scanned if the nested provenance resolution failed. It could only happen as a side-effect if another package with successful nested provenance resolution uses the same repository. Not providing a result for a package where nested provenance resolution is better in my opinion because it could give the false impression that the scan was successful. This could also be an issue with package based storages, if the incomplete result would end up in the storage.

Reply:
> I tried your approach adjusting `hasCompleteScanResult()`, but failed as that has `scanner` as second parameter which seems to me to not fit the use case.

Right, a better approach would have been to filter packages in `getNestedScanResultsByPackage` using `getNestedProvenance(id)`. I still need to look at your other PR.

Reply:
> Right, a better approach would have been to filter packages in `getNestedScanResultsByPackage` using `getNestedProvenance(id)`. I still need to look at your other PR.

That's basically what my other PR does.

Reply:
Is this PR obsolete now that #5933 is approved?

Reply:
yes, let me close it.